connectioncmd/input.go

```diff
@@ -0,0 +1,390 @@
+		}
+		m.inputs[i] = t
+	}
+	return m
+}
+
+func ApprovalInput() InputModel {
+	m := InputModel{
+		inputs:         make([]textinput.Model, 1),
+		connectionType: "Approval",
+		state:          1,
+		cursorMode:     textinput.CursorStatic,
+	}
+	t := textinput.New()
+	t.CursorStyle = selectedItemStyle
+	t.Prompt = "Your config file will be overwritten, are you want to continue? (y/n): "
+	t.Focus()
+	t.PromptStyle = noStyle
+	t.TextStyle = noStyle
+	t.SetCursorMode(textinput.CursorStatic)
+	m.inputs[0] = t
+	return m
+}
+
+func updateValues(m *InputModel) {
+	if m.state == 2 {
+		m.config.SSL.CAPath = m.inputs[0].Value()
+		m.config.SSL.CertPath = m.inputs[1].Value()
+		m.config.SSL.KeyPath = m.inputs[2].Value()
+		m.config.SSL.KeyPassword = m.inputs[3].Value()
+		m.state = 1
+	} else {
+		if m.connectionType == "Hazelcast Viridian" {
+			_ = os.Setenv("HZ_CLOUD_COORDINATOR_BASE_URL", "https://api.viridian.hazelcast.com")
+			m.config.Hazelcast.Cluster.Cloud.Enabled = true
+			m.config.Hazelcast.Cluster.Name = m.inputs[0].Value()
+			m.config.Hazelcast.Cluster.Cloud.Token = m.inputs[1].Value()
+			m.config.SSL.Enabled = true
+			m.config.SSL.ServerName = "hazelcast.cloud"
+			m.config.SSL.CAPath = m.inputs[2].Value()
+			m.config.SSL.CertPath = m.inputs[3].Value()
+			m.config.SSL.KeyPath = m.inputs[4].Value()
+			m.config.SSL.KeyPassword = m.inputs[5].Value()
+		} else if m.connectionType == "Hazelcast Cloud" {
+			_ = os.Setenv("HZ_CLOUD_COORDINATOR_BASE_URL", "https://coordinator.hazelcast.cloud")
+			m.config.Hazelcast.Cluster.Cloud.Enabled = true
+			m.config.Hazelcast.Cluster.Name = m.inputs[0].Value()
+			m.config.Hazelcast.Cluster.Cloud.Token = m.inputs[1].Value()
+			if m.inputs[2].Value() == "T" {
+				m.state = 2
+				m.config.SSL.Enabled = true
+			} else {
+				m.config.SSL.Enabled = false
+			}
+		} else if m.connectionType == "Local" {
+			m.config.Hazelcast.Cluster.Name = m.inputs[0].Value()
+			addressString := strings.ReplaceAll(m.inputs[1].Value(), " ", "")
+			m.config.Hazelcast.Cluster.Network.Addresses = strings.Split(addressString, ",")
+			if m.inputs[2].Value() == "T" {
+				m.state = 2
+				m.config.SSL.Enabled = true
+			} else {
+				m.config.SSL.Enabled = false
+			}
+		} else if m.connectionType == "Remote" {
+			m.config.Hazelcast.Cluster.Name = m.inputs[0].Value()
+			m.config.Hazelcast.Cluster.Network.Addresses = append(make([]string, 0), m.inputs[1].Value())
+			if m.inputs[2].Value() == "T" {
+				m.state = 2
+				m.config.SSL.Enabled = true
+			} else {
+				m.config.SSL.Enabled = false
+			}
+		} else if m.connectionType == "Approval" {
+			choice = m.inputs[0].Value()
+		}
+	}
+}
+
+func (m InputModel) Init() tea.Cmd {
+	return nil
+}
+
+func (m InputModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "ctrl+c", "esc":
+			m.quitting = true
+			return m, tea.Quit
+		case "tab", "enter", "up", "down":
```

Comment:
Suggestion: I think it is less error-prone to distinguish these keys similar to the following
```
switch m.Type {
		case tea.KeyTab:
                         ...
		case tea.KeyCtrlC:
			 ...
		}

Reply:
Refactored at https://github.com/hazelcast/hazelcast-commandline-client/pull/139/commits/8a912812f9624431cb43dd10992a1e2135072552